packages/nyc-test-coverage/src/index.ts

```diff
@@ -1,10 +1,51 @@
-import { InjectedSinks } from '@temporalio/worker';
+import { InjectedSinks, BundleOptions, WorkerOptions } from '@temporalio/worker';
 import { CoverageSinks } from './sinks';
 import libCoverage from 'istanbul-lib-coverage';
 
+// Pull `webpack.Configuration` type without needing to import Webpack
+type WebpackConfigType = ReturnType<NonNullable<BundleOptions['webpackConfigHook']>>;
+
 export class WorkflowCoverage {
   coverageMap = libCoverage.createCoverageMap();
 
+  /**
+   * Add all necessary coverage-specific logic to Worker config:
+   * interceptors, sinks, and Webpack config hook.
+   */
+
+  augmentWorkerOptions(workerOptions: WorkerOptions): WorkerOptions {
+    if (!workerOptions.workflowsPath) {
+      throw new TypeError('Cannot automatically instrument coverage without specifying `workflowsPath`');
+    }
```

Comment:
Could catch at compile time with:

```ts
type MakeRequired<Type, Key extends keyof Type> = Type & {
  [Property in Key]-?: Type[Property];
};

workerOptions: MakeRequired<WorkerOptions, 'workflowsPath'>
```

Reply:
We'll do this as well. I'd still prefer to keep the runtime check for JS devs.